docs/Contributing/pull-request.rst

```diff
@@ -0,0 +1,182 @@
+
+
+4. Edit Files As Needed In The Sandbox And Local Branch
+=======================================================
+
+This step is as simple as it sounds. Edit the files in the local sandbox as needed to complete the work you
+want to work on. Depending on what you are doing, there is a high likelyhood that you will want to compile
+the code to test your work. Here are some quick steps that you can use to run through quick compiles of code
+and/or documentation updates.
+
+.. code-block:: rst
+
+  ~/GitHub/freeciv21$ rm -Rf build
+  ~/GitHub/freeciv21$ cmake . -B build -G Ninja -DCMAKE_INSTALL_PREFIX=$PWD/build/install
+  ~/GitHub/freeciv21$ cmake --build build
+  ~/GitHub/freeciv21$ cmake --build build --target install
+  ~/GitHub/freeciv21$ cmake --build build --target docs
+
+
+The first command cleans out the :file:`build` directory to start fresh. The second command runs a configure
+process and then sets things up to do the install portion inside the same :file:`build` directory, keeping
+work in progress all in one place.
+
+The third and forth commands compile and installs the code. The last command is used to generate the
+documentation that you are reading right now. You can open :file:`./build/docs/index.html` to see a locally
+generated copy of the documentation prior to pushing a change up.
+
+.. note::
+  You don't have to start from the top of the list of commands every time you want to check on things. If you
+  have already done a full pass to install and then make some more changes. You can go straight to the build
+  and install commands. The :file:`cmake` program is smart enough to determine what changed and only compile
+  those files again. It's much faster to do things this way!
+
+
+One more thing to note here, this section is only talking about making changes to the Freeciv21 repository and
+has not really talked about the Games repository, nor talked about Rulesets or Tilesets. In this scenario,
+there isn't any compiling going on. The Ruleset or Tileset editor is editing files and testing locally. The
+Longturn Games repository is effectively a repository of Rulesets.
+
+
+5. Push The Changes In A Commit To Origin
+=========================================
+
+Now the moment arrives, you have a fully completed set of changes you want to share! The next step is to
+evaluate what all has changed and build a commit. A commit is an organized set of changes all bundled up
+together. There are lots of ways or organize commits, for this set of instructions we are going to assume
+that all of your changes are going to be bundled into one commit. If you want to learn how to build more than
+one commit into a Pull Request then you can read about `git add <https://git-scm.com/docs/git-add>`_ and
+`git commit <https://git-scm.com/docs/git-commit>`_. Most of the work is handled with :file:`git add`.
+
+.. code-block:: rst
+
+  ~/GitHub/freeciv21$ git status
+  ~/GitHub/freeciv21$ git add --all
+  ~/GitHub/freeciv21$ git commit
```

Comment:
This  on the other hand is needed. We could also talk about writing a good commit message (or even setup a commit message standardized template).